src/main/java/games/strategy/engine/data/properties/FileProperty.java

```diff
@@ -167,7 +166,7 @@ public String getDescription() {
         return Arrays.toString(acceptableSuffixes);
       }
     });
-    final int rVal = fileChooser.showOpenDialog(MainFrame.getInstance());
+    final int rVal = fileChooser.showOpenDialog(null);
```

Comment:
For multiple screens, it could get quite annoying, particularly if not both screens are visible at the same time. I pushed a commit to this branch that will restore the main frame reference: 0c6a9f1

Instead of a direct reference to `mainFrame` being leaked everywhere, instead `GameRunner`, which owns the `mainFrame` reference will be an intemediary for these swing calls. This way `GameRunner` can hide the `mainFrame` object reference from the rest of the system. Still pretty coupled, but a little bit less so and I hope a good step for further refactoring iterations.